### PR TITLE
Fix `make cobra-docs` target in Makefile

### LINKED ISSUE #3341: The "make cobra-docs" target no longer works
**Bug description**

During the refactoring the "make cobra-docs" target was broken.

```
$ make cobra-docs
TANZU_CLI_NO_INIT=true TANZU_CLI_NO_COLOR=true go run ./cmd/cli/tanzu generate-all-docs
go run: no packages loaded from ./cmd/cli/tanzu
make: *** [cobra-docs] Error 1
```

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- (x) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**

The target should work

**Steps to reproduce the bug**

```
cd tanzu-framework
git checkout main
make cobra-docs
```

**Version** (include the SHA if the version is not obvious)
Today's `main` branch: 9b251c3d

**Environment where the bug was observed (cloud, OS, etc)**

Locally

@prkalle, I missed that during the review of #3302

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it

This PR fixes the makefile target "cobra-docs" broken during refactoring and updated the cli command docs.

### Which issue(s) this PR fixes

Fixes #3341 

### Describe testing done for PR
Tested the  "make cobra-docs" and it generated the docs successfully.

### Release note

```release-note
N/A
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -571,9 +571,13 @@ verify-ui-bindata: ## Run verification for UI bindata
 
 .PHONY: cobra-docs
 cobra-docs:
-	TANZU_CLI_NO_INIT=true TANZU_CLI_NO_COLOR=true $(GO) run ./cmd/cli/tanzu generate-all-docs
+	#TODO: Currenyly generating docs into both cli/core/docs/cli/commands and docs/cli/commands directories. Need to change it when core CLI moved to different repo
+	cd cli/core && TANZU_CLI_NO_INIT=true TANZU_CLI_NO_COLOR=true $(GO) run ./cmd/tanzu generate-all-docs
+	sed -i.bak -E 's/\/[A-Za-z]*\/([a-z]*)\/.config\/tanzu\/pinniped\/sessions.yaml/~\/.config\/tanzu\/pinniped\/sessions.yaml/g' cli/core/docs/cli/commands/tanzu_pinniped-auth_login.md
+	cd cli/core && TANZU_CLI_NO_INIT=true TANZU_CLI_NO_COLOR=true $(GO) run ./cmd/tanzu generate-all-docs --docs-dir "../../docs/cli/commands"
```
Comment (anujc25) [inline Makefile:577]: We should not add plugin specific command details in `cli/core/docs/cli/commands`.

My preference would be not to include anything in the `cli/core` for now and let's only update in `docs/cli/commands` to fix the issue.

Comment (prkalle) [reply Makefile]: I have the same opinion. Updated.

**Review by marckhouzam — APPROVED** [review]
LGTM

Thanks @prkalle !

**Review by anujc25 — APPROVED** [review]
LGTM. Thanks

**Pull request merged**